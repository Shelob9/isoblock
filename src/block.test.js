//Fucntions to test:
import { save, edit } from "./index";
//Import a renderer to test with
import TestRenderer from "react-test-renderer";
//We will use this to make edit/save rendereable
import { createElement } from "@wordpress/element";

test("Block edit callback", () => {
  //Props with sample data
  const props = {
    attributes: {
      defaultValue: "Faces",
      blockId: ""
    },
    className: "wp-blocks-tooths",
    clientId: "something",
    setAttributes: jest.fn()
  };

  const component = TestRenderer.create(createElement(edit, props));
  expect(component.toJSON()).toMatchSnapshot();
});

test("Block save callback", () => {
  //Props with sample data
  const props = {
    attributes: {
      defaultValue: "Faces",
      blockId: ""
    },
    className: "wp-blocks-tooths",
    clientId: "something"
  };

  const component = TestRenderer.create(createElement(save, props));
  expect(component.toJSON()).toMatchSnapshot();
});

