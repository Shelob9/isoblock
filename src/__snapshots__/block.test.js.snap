// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block edit callback 1`] = `
<div>
  <form
    className="wp-blocks-tooths"
  >
    <div>
      <label
        htmlFor="something"
      >
        Form Field
      </label>
      <input
        className="isoblock-field"
        id="something"
        onChange={[Function]}
        value="Faces"
      />
    </div>
  </form>
</div>
`;

exports[`Block save callback 1`] = `
<form
  className="wp-blocks-tooths"
>
  <div>
    <label
      htmlFor=""
    >
      Form Field
    </label>
    <input
      className="isoblock-field"
      id=""
      value="Faces"
    />
  </div>
</form>
`;
